source/isaaclab/test/sensors/test_contact_sensor.py

```diff
@@ -394,6 +394,68 @@ def test_sensor_print(setup_simulation):
         print(scene.sensors["contact_sensor"])
 
 
+# minor gravity force in -z to ensure object stays on ground plane
+@pytest.mark.parametrize("grav_dir", [(-10.0, 0.0, -0.1), (0.0, -10.0, -0.1)])
+@pytest.mark.isaacsim_ci
+def test_friction_reporting(setup_simulation, grav_dir):
+    """
+    Test friction force reporting for contact sensors.
+
+    This test places a contact sensor enabled cube onto a ground plane under different gravity directions.
+    It then compares the normalized friction force dir with the direction of gravity to ensure they are aligned.
+    """
+    sim_dt, _, _, _, carb_settings_iface = setup_simulation
+    carb_settings_iface.set_bool("/physics/disableContactProcessing", True)
+    device = "cuda:0"
+    sim_cfg = SimulationCfg(dt=sim_dt, device=device, gravity=grav_dir)
+    with build_simulation_context(sim_cfg=sim_cfg, add_lighting=False) as sim:
+        sim._app_control_on_stop_handle = None
+
+        scene_cfg = ContactSensorSceneCfg(num_envs=1, env_spacing=1.0, lazy_sensor_update=False)
+        scene_cfg.terrain = FLAT_TERRAIN_CFG
+        scene_cfg.shape = CUBE_CFG
+
+        filter_prim_paths_expr = [scene_cfg.terrain.prim_path + "/terrain/GroundPlane/CollisionPlane"]
+
+        scene_cfg.contact_sensor = ContactSensorCfg(
+            prim_path=scene_cfg.shape.prim_path,
+            track_pose=True,
+            debug_vis=False,
+            update_period=0.0,
+            track_air_time=True,
+            history_length=3,
+            track_friction_forces=True,
+            filter_prim_paths_expr=filter_prim_paths_expr,
+        )
+
+        scene = InteractiveScene(scene_cfg)
+
+        sim.reset()
+
+        scene["contact_sensor"].reset()
+        scene["shape"].write_root_pose_to_sim(
+            root_pose=torch.tensor([0, 0.0, CUBE_CFG.spawn.size[2] / 2.0, 1, 0, 0, 0])
+        )
+
+        # step sim once to compute friction forces
+        _perform_sim_step(sim, scene, sim_dt)
+
+        # check that forces are being reported match expected friction forces
+        expected_friction, _, _, _ = scene["contact_sensor"].contact_physx_view.get_friction_data(dt=sim_dt)
+        reported_friction = scene["contact_sensor"].data.friction_forces_w[0, 0, :]
+
+        assert torch.allclose(expected_friction.sum(dim=0), reported_friction[0], atol=1e-6)
+
+        grav = torch.tensor(grav_dir, device=device)
```

Comment:
Please add comments in the code to make them easier to understand without having to read through the code explicitly.